Talker demo node fails when run with rmw_opensplice_cpp

<!--
For general questions, please ask on ROS answers: https://answers.ros.org, make sure to include at least the `ros2` tag and the rosdistro version you are running, e.g. `ardent`.
For general design discussions, please post on discourse: https://discourse.ros.org/c/ng-ros
Not sure if this is the right repository? Open an issue on https://github.com/ros2/ros2/issues
For Bug report or feature requests, please fill out the relevant category below
-->

## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 18.04
- Installation type:
  - source
- Version or commit hash:
  - master branch 2ef6b61710f7095caad7cb5971bc6348fc1869bf
- DDS implementation:
  - rmw_opensplice
- Client library (if applicable):
  - rclcpp

#### Steps to reproduce issue
<!-- Detailed instructions on how to reliably reproduce this issue http://sscce.org/
``` code that can be copy-pasted is preferred ``` -->
Follow directions for building ROS2 from source, and working with multiple rmw implementations: https://index.ros.org//doc/ros2/Tutorials/Working-with-multiple-RMW-implementations/
```
mkhansen@mkhansen-desk:~/ros2_dev/ros2_ws/src/ros2/rmw_opensplice$ export RMW_IMPLEMENTATION=rmw_opensplice_cpp
mkhansen@mkhansen-desk:~/ros2_dev/ros2_ws/src/ros2/rmw_opensplice$ ros2 run demo_nodes_cpp talker
```

#### Expected behavior
"Hello World " should be printing on the screen

#### Actual behavior
```
========================================================================================
Context     : DDS::DomainParticipantFactory::create_participant
Date        : 2019-09-12T13:29:32-0700
Node        : mkhansen-desk
Process     : talker <6636>
Thread      : main thread 7f5596336f40
Internals   : DomainParticipantFactory.cpp/324/6.9.190705OSS///-1
----------------------------------------------------------------------------------------
Report      : Unable to connect to domain id = 7.
              The most common causes of this error are an incorrect configuration file or
              that OpenSpliceDDS is not running (when using shared memory mode).
Internals   : u_participantNew/u_participant.c/234/774/1568320172.795758339
----------------------------------------------------------------------------------------
Report      : Failed to initialize domain configuration for domain 7
Internals   : u__domainOpen/u_domain.c/1485/774/1568320172.795762902
----------------------------------------------------------------------------------------
Report      : Invalid URI specified: "file:///share/opensplice_cmake_module/config/ros_ospl.xml".
Internals   : user::u_domain::GetDomainConfig/u_domain.c/241/774/1568320172.795765966

>>> [rcutils|error_handling.c:107] rcutils_set_error_state()
This error state is being overwritten:

  'failed to create domain participant, at /home/mkhansen/ros2_dev/ros2_ws/src/ros2/rmw_opensplice/rmw_opensplice_cpp/src/rmw_node.cpp:162, at /home/mkhansen/ros2_dev/ros2_ws/src/ros2/rcl/rcl/src/rcl/node.c:326'

with this new error message:

  'rcl node's rmw handle is invalid, at /home/mkhansen/ros2_dev/ros2_ws/src/ros2/rcl/rcl/src/rcl/node.c:464'

rcutils_reset_error() should be called after error handling to avoid this.
<<<
[ERROR] [rcl]: Failed to fini publisher for node: 1
terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
  what():  failed to initialize rcl node: rcl node's rmw handle is invalid, at /home/mkhansen/ros2_dev/ros2_ws/src/ros2/rcl/rcl/src/rcl/node.c:464
```
#### Additional information
This is new behavior, I've run opensplice successfully in the past on this machine. Most recently was 8/29/19. Also I asked another member of the team to try the same demo and he got the same error.

--- a/opensplice_cmake_module/env_hook/opensplice.sh.in
+++ b/opensplice_cmake_module/env_hook/opensplice.sh.in
@@ -33,6 +33,16 @@ else
   _OSPL_HOME_TO_USE=$OSPL_HOME
 fi
 
+# TODO(jacobperron): Find a better solution that does not rely on colcon
+# if AMENT_CURRENT_PREFIX is not set, fall back to destination set at configure time
+if [ ! -d "$AMENT_CURRENT_PREFIX" ]; then
+  if [ -z "$COLCON_CURRENT_PREFIX" ]; then
+    echo "The compile time prefix path '$AMENT_CURRENT_PREFIX' doesn't " \
+      "exist. Consider sourcing a different extension than '.sh'." 1>&2
+  else
+    AMENT_CURRENT_PREFIX="$COLCON_CURRENT_PREFIX"
+  fi
+fi
 
 _DEFAULT_OSPL_URI="file://$AMENT_CURRENT_PREFIX/share/opensplice_cmake_module/config/ros_ospl.xml"
 

--- a/opensplice_cmake_module/env_hook/opensplice.sh.in
+++ b/opensplice_cmake_module/env_hook/opensplice.sh.in
@@ -34,7 +34,7 @@ else
 fi
 
 # TODO(jacobperron): Find a better solution that does not rely on colcon
-# if AMENT_CURRENT_PREFIX is not set, fall back to destination set at configure time
+# if AMENT_CURRENT_PREFIX is not set, try to use COLCON_CURRENT_PREFIX if available
 if [ ! -d "$AMENT_CURRENT_PREFIX" ]; then
   if [ -z "$COLCON_CURRENT_PREFIX" ]; then
     echo "The compile time prefix path '$AMENT_CURRENT_PREFIX' doesn't " \

